pandas/tests/series/indexing/test_indexing.py

```diff
@@ -366,6 +367,14 @@ def test_loc_setitem_nested_data_enlargement():
     tm.assert_series_equal(ser, expected)
 
 
+def test_loc_ea_numeric_index_oob_slice_end():
```

Comment:
Could you also add a test for a pyarrow-backed numeric type?

Reply:
This would fail, ``int64[pyarrow]`` returns false in ``is_numeric_dtype``. Is this expected?  Index.is_numeric will be deprecated, so should not use that either

Reply:
Ah. I think all the dtype checkers should become pyarrow aware eventually but okay not to tackle that in this PR